sdv/single_table/base.py

```diff
@@ -500,6 +506,75 @@ def get_info(self):
 
         return info
 
+    def _fit_constraint_column_formatters(self, data):
+        """Fit formatters for columns that are dropped by constraints before data processing."""
+        self._constraint_col_formatters = {}
+        self._constraint_col_dtypes = {}
```

Comment:
Is it required since we set them in the `__init__`  also?

Reply:
Oh good catch, I forgot we added fixing the dtypes to the CAG base class. I removed the dtypes conversion in the base synthesizer, so now it should only be learning/applying formatters to the dropped columns.